teps/0061-allow-custom-task-to-be-embedded-in-pipeline.md

```diff
@@ -356,6 +365,39 @@ add them under "/teps/images/". It's upto the TEP author to choose the name
 of the file, but general guidance is to include at least TEP number in the
 file name, for example, "/teps/images/NNNN-workflow.jpg".
 -->
+The actual code changes needed to implement this TEP, are very minimal.
+
+**Broad categories are:**
+1. Add the relevant APIs. 
+   Already covered in [Proposal section](#proposal).
+   
+2. Change validation logic to accept the newly added API fields.
+    Currently `tecktoncd/pipeline` will reject any request for `Run`, 
+    which does not include a `Run.RunSpec.Ref`. So this validation is now changed to
+    either one of `Ref` or `Spec` must be present, but not both. 
+    
+    Next, whether it is a `Ref` or a `Spec`, validation logic will ensure, they have 
+    non-empty values for, `APIVersion` and `Kind`.
+    
+    Lastly document advice for downstream custom controllers to implement their
+    own validation logic. This aspect is covered in full detail, in
+    [Upgrade &amp; Migration Strategy section](#upgrade--migration-strategy-optional)
+    of this TEP.
+
+This TEP does not change the existing flow of creation of `Run` object, it updates 
+the Run object with the content of `RunSpec.Spec` by marshalling the field 
+`Spec runtime.RawExtension` to json and embed in the spec, before creating the
+Run object.
+
+Code changes maybe implemented in three PRs.
+
+1. *Core changes*, i.e. Addition of APIs as per approved proposal, change validations and update
+unit tests for testing new APIs and validation logic.
+   
+2. *e2e tests* Add an e2e test for testing a pipelineRun with custom task embedded.
+
+3. *documentation* Update documentation on custom tasks, to include embedded spec. Also add the 
+    guide on validation for downstream custom task developers.
```

Comment:
I would recommend to have it split in two PRs rather than three. (1) Add documentation with a `preview` mode (2) Code changes + testing

At the same time, it's great to have it but does not have to be part of the TEP.

Reply:
This sounds good to me !